torchrec/csrc/dynamic_embedding/details/io.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+#include <torch/torch.h>
+#include <torchrec/csrc/dynamic_embedding/details/io_registry.h>
+#include <cstdint>
+#include <span>
+
+namespace tde::details {
+
+class IO {
+ public:
+  explicit IO(const std::string& config);
+  ~IO();
+
+  IO(const IO&) = delete;
+  IO& operator=(const IO&) = delete;
+  IO(IO&&) noexcept = delete;
+  IO& operator=(IO&&) noexcept = delete;
+
+  /**
+   * Fetch parameter and optimizer states from ParamServer.
+   * @param global_ids global ids to fetch
+   * @param num_optimizer_states number of optimizer stats to fetch
+   * @param type data type
+   * @param on_fetch_complete  fetch complete callback. The parameter is
+   * a vector, the vector's size is equal to global_ids.size() *
+   * max(col_ids.size(), 1). If the parameter server does not contains some
+   * parameter, the tensor will be empty. Also, the tensor shape is
+   * [num_optimizer_states, embedding_size]. The shape of each global id can be
+   * different in some algorithm.
+   *
+   * @note this method is asynchronous, The col_ids, and global_ids will be
+   * copied inside, so it is safe to free col_ids/global_ids before
+   * on_fetch_complete.
+   */
+  void pull(
+      const std::string& table_name,
+      std::span<const int64_t> global_ids,
+      std::span<const int64_t> col_ids,
```

Comment:
Added. `col_ids` is the column id of the embedding. Because an embedding will be sharded with column-wise sharding, we hope to store different segments as different kv pairs in PS. And the column id represents the index of the column segment.